skymp5-server/cpp/lib_espm/espm.cpp

```diff
@@ -1074,21 +1074,31 @@ espm::Effects::Data espm::Effects::GetData(
   }
 
   Data result;
-  uint32_t effect = 0;
+  uint32_t effectIndex = 0;
+  bool orderFlag = true;
+  bool isValid = true;
 
   espm::RecordHeaderAccess::IterateFields(
     parent,
     [&](const char* type, uint32_t size, const char* data) {
       if (!memcmp(type, "EFID", 4)) {
+        isValid = orderFlag == true;
         result.effects.emplace_back();
-        result.effects[effect].effectId =
+        result.effects[effectIndex].effectId =
           *reinterpret_cast<const uint32_t*>(data);
+        orderFlag = false;
       } else if (!memcmp(type, "EFIT", 4)) {
-        Effect& eff = result.effects[effect];
+        isValid = orderFlag == false;
+        Effect& eff = result.effects[effectIndex];
         eff.magnitude = *reinterpret_cast<const float*>(data);
         eff.areaOfEffect = *reinterpret_cast<const uint32_t*>(data + 4);
         eff.duration = *reinterpret_cast<const uint32_t*>(data + 8);
-        effect++;
+        effectIndex++;
+        orderFlag = true;
+      }
+      if (!isValid) {
```

Comment:
It's also not valid if in the end `orderFlag` is `false`

Reply:
Hmm, I thought `IterateFields` has support for stopping iterating, but I can't see any... would be a good way to check this, but for now let's duplicate this in the end

Reply:
By the name of @SupAidmi I force this PR to be accepted